File: Kalorie/Kalorie/Core/Utils/Log.swift
//
//  Log.swift
//  Kalorie
//
//  Created by Josef Antoni on 30.08.2026.
//

import FirebaseCrashlytics
import OSLog

enum Log {

    // MARK: - Functions

    static func warning(_ error: Error, category: String = "app") {
        logger(category).warning("\(String(describing: error), privacy: .public)")
    }

    static func error(_ error: Error, category: String = "app") {
        logger(category).error("\(String(describing: error), privacy: .public)")
        Crashlytics.crashlytics().record(error: error)
    }

    static func logger(_ category: String) -> Logger {
        Logger(subsystem: "antoni.Kalorie", category: category)
    }
}

File: Android/app/src/main/java/antoni/kalorie/core/utils/Log.kt
package antoni.kalorie.core.utils

import android.util.Log as AndroidLog

fun interface ErrorReporting {
    fun record(error: Throwable)
}

object Log {

    // MARK: - Properties

    @Volatile
    var errorReporting: ErrorReporting = ErrorReporting { }

    // MARK: - Functions

    fun warning(error: Throwable, category: String = "app") {
        AndroidLog.w(category, error.toString())
    }

    fun error(error: Throwable, category: String = "app") {
        AndroidLog.e(category, error.toString())
        errorReporting.record(error)
    }
}
